data/helpers.d/utils

```diff
@@ -24,3 +24,62 @@ ynh_get_plain_key() {
         fi
     done
 }
+
+# Restore a previous backup if the upgrade process failed
+#
+# usage:
+# ynh_backup_before_upgrade
+# ynh_clean_setup () {
+# 	ynh_backup_after_failed_upgrade
+# }
+# ynh_abort_if_errors
+#
+ynh_backup_after_failed_upgrade () {
+	echo "Upgrade failed." >&2
+	app_bck=${app//_/-}	# Replace all '_' by '-'
+    # Check if a existing backup can be found before remove and restore the application.
```

Comment:
Typos :  Check if a**n** existing backup can be found before remov**ing** and restor**ing** the application.